lib/openstack_query/query_blocks/query_builder.py

```diff
@@ -123,14 +141,70 @@ def parse_where(
             )
         else:
             logger.info(
-                "Found server-side filters for preset '%s': prop '%s' pair: {%s}",
+                "Found %s set of server-side filters for preset '%s': prop '%s' pair",
+                len(server_side_filters),
                 preset.name,
                 prop.name,
-                ", ".join(
-                    [f"{key}: '{val}'" for key, val in self.server_side_filters.items()]
-                ),
             )
 
+        logger.debug(
+            "server-side-filters found: %s",
+            {f"{server_side_filters}" if server_side_filters else "None"},
+        )
+        logger.debug(
+            "client-side-filter found: %s",
+            {client_side_filter if client_side_filter else "None"},
+        )
+
+        self._add_filter(
+            client_side_filter=client_side_filter,
+            server_side_filters=server_side_filters,
+        )
+
+    def _add_filter(
+        self,
+        client_side_filter: ClientSideFilterFunc,
```

Comment:
What if there's no ClientSideFilters? Should this be optional too?

Reply:
nope - there's always a client-side filter. Otherwise we raise an error when we try and get the filter